cumulusci/tasks/push/tests/test_push_tasks.py

```diff
@@ -0,0 +1,481 @@
+        status="Succeeded",
+        sf_id=SF_ID,
+    )
+
+
+@pytest.fixture
+def package_push_job_failure():
+    return PackagePushJob(
+        push_api=mock.MagicMock(),
+        request=mock.Mock(),
+        org="00D63000000ApoXEAS",
+        status="Failed",
+        sf_id=SF_ID,
+    )
+
+
+@pytest.fixture
+def package_push_job_cancel():
+    return PackagePushJob(
+        push_api=mock.MagicMock(),
+        request=mock.Mock(),
+        org="00D63000000ApoXEAS",
+        status="Canceled",
+        sf_id=SF_ID,
+    )
+
+
+@pytest.fixture
+def package_push_request_failure():
+    return PackagePushRequest(
+        push_api=mock.MagicMock(),
+        version="1.2.3",
+        start_time="12:03",
+        status="Failed",
+        sf_id=SF_ID,
+    )
+
+
+@pytest.fixture
+def package_push_request_cancel():
+    return PackagePushRequest(
+        push_api=mock.MagicMock(),
+        version="1.2.3",
+        start_time="12:03",
+        status="Canceled",
+        sf_id=SF_ID,
+    )
+
+
+def test_parse_version():
+    task = create_task(BaseSalesforcePushTask, options={})
+    assert task._parse_version("1.2") == {
+        "major": "1",
+        "minor": "2",
+        "patch": None,
+        "build": None,
+        "state": "Released",
+    }
+    assert task._parse_version("1.2,Beta 3") == {
+        "major": "1",
+        "minor": "2",
+        "patch": None,
+        "build": "Beta 3",
+        "state": "Beta",
+    }
+    assert task._parse_version("1.2,(Beta 3)") == {
+        "major": "1",
+        "minor": "2",
+        "patch": None,
+        "build": "(Beta 3",
+        "state": "Beta",
+    }
+    assert task._parse_version("1.2") == {
+        "major": "1",
+        "minor": "2",
+        "patch": None,
+        "build": None,
+        "state": "Released",
+    }
+
+
+def test_get_version():
+    task = create_task(BaseSalesforcePushTask, options={})
+    assert task._get_version(
+        MetadataPackage(
+            push_api=mock.MagicMock(), name=NAME, sf_id=SF_ID, namespace=NAMESPACE
+        ),
+        "1.2.3.4",
+    )
+    assert task._get_version(
+        MetadataPackage(
+            push_api=mock.MagicMock(), name=NAME, sf_id=SF_ID, namespace=NAMESPACE
+        ),
+        "1.2,Beta 3",
+    )
+
+
+def test_get_version_error():
+    package = mock.MagicMock()
+    package.get_package_version_objs.return_value = None
+    task = create_task(BaseSalesforcePushTask, options={})
+    with pytest.raises(PushApiObjectNotFound):
+        task._get_version(package, VERSION)
+
+
+def test_get_package():
+    task = create_task(BaseSalesforcePushTask, options={})
+    task.push = mock.MagicMock()
+    task.push.get_package_objs.return_value = [
+        {"push_api": "123", "sf_id": SF_ID, "name": NAME, "namespace": NAMESPACE}
+    ]
+    assert task._get_package(NAMESPACE) == {
+        "push_api": "123",
+        "sf_id": SF_ID,
+        "name": NAME,
+        "namespace": NAMESPACE,
+    }
```

Comment:
The _real_ SalesforcePushApi.get_package_objs returns a list of MetadataPackage instances, not a list of dicts. For this test really all we need to prove is that BaseSalesforcePushTask._get_package returns the first item from the list that was returned by SalesforcePushApi.get_package_objs. So you could do one of two things:
1. Change this to build a MetadataPackage instance for the faked SalesforcePushApi.get_package_objs return value
2. Create an arbitrary object, `expected_package = object()`. Make SalesforcePushApi.get_package_objs return a list including that object. Assert that BaseSalesforcePushTask._get_package returns the same object (`assert task._get_package(NAMESPACE) is expected_package`)